site/pages/create-disruption/[disruptionId].page.tsx

```diff
@@ -105,22 +124,43 @@ const CreateDisruption = (props: DisruptionPageProps): ReactElement => {
 
     const getValidityRows = () => {
         if (pageState.inputs.validity) {
-            return pageState.inputs.validity.map((validity, i) => ({
-                header: `Validity period ${i + 1}`,
-                cells: [
-                    validity.disruptionEndDate && validity.disruptionEndTime && !validity.disruptionNoEndDateTime
-                        ? `${validity.disruptionStartDate} ${validity.disruptionStartTime} - ${validity.disruptionEndDate} ${validity.disruptionEndTime}`
-                        : `${validity.disruptionStartDate} ${validity.disruptionStartTime} - No end date/time`,
-                    <button
-                        id={`remove-validity-period-${i + 1}`}
-                        key={`remove-validity-period-${i + 1}`}
-                        className="govuk-link"
-                        onClick={(e) => removeValidity(e, i)}
-                    >
-                        Remove
-                    </button>,
-                ],
-            }));
+            return pageState.inputs.validity.map((validity, i) => {
+                const endingOnDate =
+                    validity.disruptionRepeats === "daily"
+                        ? validity.disruptionDailyRepeatsEndDate
+                        : validity.disruptionWeeklyRepeatsEndDate;
+
+                return {
+                    header: `Validity period ${i + 1}`,
+                    cells: [
+                        validity.disruptionEndDate &&
+                        validity.disruptionEndTime &&
+                        !validity.disruptionNoEndDateTime ? (
+                            validity.disruptionRepeats !== "doesntRepeat" &&
+                            validity.disruptionRepeats &&
```

Comment:
We should be checking if disruptionRepeats before checking if it equals something.

Reply:
done